plugins/languages/treesitter/treesitter.nix

```
@@ -142,12 +142,17 @@ helpers.neovim-plugin.mkNeovimPlugin config {
       List of parsers to ignore installing. Used when `ensure_installed` is set to "all".
     '';
 
-    parser_install_dir = helpers.defaultNullOpts.mkStr "$XDG_DATA_HOME/nvim/treesitter" ''
-      Location of the parsers to be installed by the plugin (only needed when `nixGrammars` is disabled).
+    parser_install_dir = mkOption {
+      type = with types; nullOr str;
+      default = "$XDG_DATA_HOME/nvim/treesitter";
+      defaultText = "$XDG_DATA_HOME/nvim/treesitter";
+      description = ''
+        Location of the parsers to be installed by the plugin (only needed when `nixGrammars` is disabled).
 
-      This default might not work on your own install, please make sure that `$XDG_DATA_HOME` is set if you want to use the default.
-      Otherwise, change it to something that will work for you!
-    '';
+        This default might not work on your own install, please make sure that `$XDG_DATA_HOME` is set if you want to use the default.
+        Otherwise, change it to something that will work for you!
```

Depending on the final implementation, this note will likely need re-writing.

---

```suggestion

        If left as default, parsers are installed to the "site" dir.

        If set to null the _plugin default_ is used, which will not work on nix.

        If a custom path is used it must be added to the `runtimepath`, e.g. using:
        ```lua
        vim.opt.runtimepath:append('/some/path/to/store/parsers')
        ```
```